design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,280 @@
+| Azure Instance Metadata Service (IMDS) | a REST endpoint accessible to all VMs created via the Azure Resource Manager. Resources in Azure request access tokens from the Azure Instance Metadata service endpoint                        |
+| JWT                                    | JSON-based access token that asserts claims (logged in, etc)         
+
+## Useful links
+
+- [Azure Spike Ramp-up training exercise](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Azure+Spike+Ramp-up#AzureSpikeRamp-up-/)
+
+- [Delivery Plan - Conjur Azure Authenticator](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Delivery+Plan+-+Conjur+Azure+Authenticator)
+
+## Issue description
+
+Many of our customers are using Azure as their main cloud provider. Therefore we want to enable Azure resources to authenticate according to Azure 
+properties and get back a Conjur access token in order to retrieve secrets from our providers. At current, customers can authenticate their Azure 
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the feature doc, before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+    - !variable
+        id: provider-uri # https://sts.windows.net/TENANT_ID
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+### Azure Application Identity
+
+To authenticate with Conjur, Azure-specific fields will need to be provided in the host annotations of the Conjur host identity for the Azure resource, 
+specifically subscription-id, resource-group, and user/system-assigned-identity. These fields will be used to confirm the identity of the Azure resource 
+sending the authentication request by comparing the Azure token information with the host annotations that are defined in the above policy.
+
+A Conjur host will be defined as follows with their annotations holding Azure-specific identification properties.
+
+```yaml
+- !policy 
+  id: azure-apps 
+  body: 
+  - !host 
+    id: test-app 
+    annotations: 
+    authn-azure/subscription-id: test-subscription 
+    authn-azure/resource-group: test-group 
+    authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+    authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
```

Comment:
I would change it to a real system assigned identity (which is GUID) to avoid confusion

Reply:
why would it add confusion? b/w what is in the azure token (guid) and how we reference it here?

Reply:
I think that @shaharglazner means that the example should be a GUID. Something like:
```
authn-azure/system-assigned-identity: 853b9a84-5bfa-4b22-a3f3-0b9a43d9ad8a
```

Reply:
I updated this in the doc